ethcontract/src/transaction/gas_price.rs

```diff
@@ -1,121 +1,60 @@
 //! Implementation of gas price estimation.
 
-use crate::errors::ExecutionError;
 use primitive_types::U256;
-use web3::api::Web3;
-use web3::Transport;
+use web3::types::U64;
 
 /// The gas price setting to use.
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum GasPrice {
-    /// The standard estimated gas price from the node, this is usually the
-    /// median gas price from the last few blocks. This is the default gas price
-    /// used by transactions.
-    Standard,
-    /// A factor of the estimated gas price from the node. `GasPrice::Standard`
-    /// is similar to `GasPrice::Scaled(1.0)` but because of how the scaling is
-    /// calculated, `GasPrice::Scaled(1.0)` can lead to some rounding errors
-    /// caused by converting the estimated gas price from the node to a `f64`
-    /// and back.
-    Scaled(f64),
-    /// Specify a specific gas price to use for the transaction. This will cause
-    /// the transaction `SendFuture` to not query the node for a gas price
-    /// estimation.
-    Value(U256),
+    /// Legacy type of transactions, using single gas price value. Equivalent to sending
+    /// eip1559 transaction with max_fee_per_gas = max_priority_fee_per_gas = gas_price
+    Legacy(U256),
+
+    /// Eip1559 type of transactions, using two values (max_fee_per_gas, max_priority_fee_per_gas)
+    Eip1559((U256, U256)),
```

Comment:
nit: It would be nice to use names so that there are no confusions about the order:
```suggestion
    Eip1559 {
        max_fee_per_gas: U256,
        max_priority_fee_per_gas: U256,
    }
```

Reply:
Agreed. Fixed.